### DISCUSSION TITLE
Push notifications for Android

### DISCUSSION THREAD

**Discussion by maxsilnov** [dodorare/crossbow#158] [Q&A] [answered]
Hi! Is it possible to send notifications on android ? Or where can I read about it ?

**Comment by enfipy** [comment, marked as answer]
Hey! Currently, notifications are unavailable, but I think you can pretty easily create a plugin and make it possible! Of course, it depends on what provider you want to use, etc.

To implement a custom plugin with this functionality - you will need to know the following things:
1. How to create a Crossbow plugin for Android to handle notifications: [link](https://crossbow.dodorare.com/crossbow/android-plugins.html).
2. How to collect UID and remove UID when uninstalling: [link](https://developers.google.com/cloud-messaging/registration).
3. How to handle client notifications: [link](https://developers.google.com/cloud-messaging/android/client).
4. How to send the notification message: [link](https://developers.google.com/cloud-messaging/downstream).

Also, there's an option to adapt some Godot plugins (as Crossbow uses very similar design approach) for [godot-firebase](https://github.com/FrogSquare/GDFirebase) or [godot-local-notifications](https://github.com/DrMoriarty/godot-local-notification).

P.S: We would love to add this functionality - so any ideas will be highly appreciated.
